api4/system.go

```diff
@@ -64,11 +65,32 @@ func getSystemPing(c *Context, w http.ResponseWriter, r *http.Request) {
 	if r.FormValue("get_server_status") != "" {
 		dbStatusKey := "database_status"
 		s[dbStatusKey] = model.STATUS_OK
-		_, appErr := c.App.Srv.Store.System().Get()
-		if appErr != nil {
-			mlog.Debug(fmt.Sprintf("Unable to get database status: %s", appErr.Error()))
+
+		// Database Write/Read Check
+		currentTime := fmt.Sprintf("%d", time.Now().Unix())
+		healthCheckKey := "health_check"
+
+		writeErr := c.App.Srv.Store.System().SaveOrUpdate(&model.System{
+			Name:  healthCheckKey,
+			Value: currentTime,
+		})
+		if writeErr != nil {
+			mlog.Debug(fmt.Sprintf("unable to write to database: %s", writeErr.Error()))
 			s[dbStatusKey] = model.STATUS_UNHEALTHY
 			s[model.STATUS] = model.STATUS_UNHEALTHY
+		} else {
+			healthCheck, readErr := c.App.Srv.Store.System().GetByName(healthCheckKey)
+			if readErr != nil {
+				mlog.Debug(fmt.Sprintf("unable to read from database: %s", readErr.Error()))
+				s[dbStatusKey] = model.STATUS_UNHEALTHY
+				s[model.STATUS] = model.STATUS_UNHEALTHY
+			} else if healthCheck.Value != currentTime {
+				mlog.Debug(fmt.Sprintf("incorrect healthcheck value, expected %s, got %s", healthCheck.Value, currentTime))
```

Comment:
It looks like we need to switch around the expected + actual values here.

```suggestion
				mlog.Debug(fmt.Sprintf("Incorrect healthcheck value: expected %s, got %s", currentTime, healthCheck.Value))
```

Reply:
Yes, good catch.